File: 은지/LV2/튜플.py
def solution(s):
    answer = []
    n_lst = [0]*100001
    
    num = ""
    for i in range(len(s)):
        if s[i] != "}" and s[i] != "{" and s[i] != ",":
            num += s[i]
        elif len(num):
            n_lst[int(num)] += 1
            num = ""        

    lst = []
    for i in range(100001):
        if n_lst[i]:
            lst.append((n_lst[i], i))
    lst.sort(reverse=True)
    for x in lst:
        answer.append(x[1])
    return answer

s = "{{20,111},{111}}"
print(solution(s))

File: 은지/LV2/튜플.js
function solution(s) {
  var answer = [];
  var n_lst = []
  for (let i=0; i<=100001; i++) {
    n_lst.push(0)
  }
  var num = ""
  for (let i=0; i<s.length; i++) {
    if (s[i] !== "{" && s[i] !== "}" && s[i] !== ",") {
      num += s[i]
    } else if (num.length) {
      n_lst[Number(num)] += 1
      num = ""
    }
  }

  var lst = []
  for (let i=0; i<100001; i++) {
    if (n_lst[i]) {
      lst.push([n_lst[i], i])
    }
  }
  lst.sort((a, b) => b[0]-a[0])
  for (let x of lst) {
    answer.push(x[1])
  }
  return answer;
}

s = "{{20,111},{111}}"
console.log(solution(s))